web/frontend/src/pages/election/Result.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { FC, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { RankResults, SelectResults, TextResults } from 'types/electionInfo';
+import SelectResult from './components/SelectResult';
+import RankResult from './components/RankResult';
+import TextResult from './components/TextResult';
+import {
+  ID,
+  RANK,
+  RankQuestion,
+  SELECT,
+  SUBJECT,
+  SelectQuestion,
+  Subject,
+  SubjectElement,
+  TEXT,
+} from 'types/configuration';
+import DownloadButton from 'components/buttons/DownloadButton';
+import { useTranslation } from 'react-i18next';
+import saveAs from 'file-saver';
+import { ROUTE_ELECTION_INDEX } from 'Routes';
+import { Link, useParams } from 'react-router-dom';
+import TextButton from '../../components/buttons/TextButton';
+import useElection from 'components/utils/useElection';
+import { useConfigurationOnly } from 'components/utils/useConfiguration';
+
+// Functional component that display the result of the votes
+
+const ElectionResult: FC = () => {
+  const { t } = useTranslation();
+  const { electionId } = useParams();
+
+  const { loading, result, configObj } = useElection(electionId);
+  const configuration = useConfigurationOnly(configObj);
+
+  const [rankResult, setRankResult] = useState<RankResults>(null);
+  const [selectResult, setSelectResult] = useState<SelectResults>(null);
+  const [textResult, setTextResult] = useState<TextResults>(null);
+
+  // Group the different results by the ID of the question SelectResult
+  // are mapped to 0 or 1s, such that ballots can be count more efficiently
+  const groupByID = (
+    resultMap: Map<any, any>,
```

Comment:
Looks great overall ! I have a reservation about the use of  the`any` type though. Can't we use `Map<ID, number[][]>` for the map ?
Same for `results`, it seems that it should be `string[][] | number[][] | boolean[][]`.

Reply:
The answer is pretty simple, I forgot we could type objects with ORs ! 😔